lib/src/extractLevels.js

```diff
@@ -0,0 +1,191 @@
+  const inputIfcFilename = getArgInputFilename()
+  const relElementsGrouped = await extractRELID(inputIfcFilename)
+  await createManipulateNewIFC(inputIfcFilename, relElementsGrouped)
+}
+
+
+/** @return {string} filename from command line. */
+export function getArgInputFilename() {
+  const args = process.argv.slice(2)
+  if (args.length < 1) {
+    console.error(getUsage())
+    return
+  }
+  const ifcFilename = args[0]
+  console.log('Input file: ', ifcFilename)
+  return ifcFilename
+}
+
+
+/**
+ * Open IFC model and extract related elements.
+ * @param {string} ifcFilename
+ * @return {array} related elements
+ */
+export async function extractRELID(ifcFilename) {
+  const model = new IfcModel()
+  const rawFileData = fs.readFileSync(ifcFilename)
+  await model.open(rawFileData)
+
+  const ifcRelSpatial = model.getEltsOfNamedType('IFCRELCONTAINEDINSPATIALSTRUCTURE')
+  const ifcSpace = model.getEltsOfNamedType('IFCSPACE')
+  const AllrelElementsArr = []
+
+  for (let i = 0; i< ifcRelSpatial.length; i++) {
+    const relElements = ifcRelSpatial[i].RelatedElements
+    const relElementsArr = []
+
+    for (let j = 0; j < relElements.length; j++) {
+      relElementsArr[j] = relElements[j].value
+    }
+    AllrelElementsArr[i] = relElementsArr
+  }
+  for (let i = 0; i< ifcSpace.length; i++) {
+    IFCSpacesELementsArrAll[i] = ifcSpace[i].expressID
+  }
+  const relElementsGroupedAll = groupElements(AllrelElementsArr)
+  return relElementsGroupedAll
+}
+
+
+/**
+ * Group related element arrays for removal of each level
+ * @param {array} elementArray Extracted related element arrays from IFC file
+ * @return {array} groupedElements
+ */
+function groupElements(elementArray) {
+  const groupedElements = []
+  for (let i = 0; i<elementArray.length+1; i++) {
+    let joinedArray = []
+    joinedArray = elementArray
+    joinedArray.splice(i, 1)
+    const flatElements = joinedArray.flat()
+    groupedElements[i] = flatElements
+  }
+  return groupedElements
+}
+
+
+/**
+ * Create a set of new IFC files and remove related elements fot each level
+ * @param {string} inputIfcFilename
+ * @param {array} relatedElemGrouped Express IDs of grouped related
+ *     elements for each level
+ */
+function createManipulateNewIFC(inputIfcFilename, relatedElemGrouped) {
+  for (let i = 0; i < relatedElemGrouped.length; i++) {
+    const newfilename = `${inputIfcFilename}_Level-${i}.ifc`
+    copyFiles(inputIfcFilename, newfilename, relatedElemGrouped[i])
+  }
+}
+
+
+/**
+ * Create map to find Line numbers based on expressID
+ * @param {string} IFCdata - Input raw IFC data
+ * @return {Map} Map with Express IDs as keys
+ */
+function expressIDtoLineMap(IFCdata) {
+  const expressIDMaps = new Map()
+  const lines = IFCdata.split('\n')
+  for (let i = 0; i < lines.length; i++) {
+    const line = lines[i]
+    {
+      if ((line.indexOf('#') != -1) && (line.indexOf('=') != -1)) {
+        const data1 = line.split('#')
```

Comment:
You sure this is right?  hashes both define elements and are used for refs, e.g.:
```
#5=IFCAPPLICATION(#1,'2021','Autodesk Revit 2021 (ENU)','Revit');
```

Reply:
Yes. The next line splits uing "=" to make sure its not targeting the refs